packages/core/src/app/payment/Payment.tsx

```diff
@@ -370,7 +370,7 @@ class Payment extends Component<
                 window.location.reload();
             }
 
-            if (isErrorWithType(error)) {
+            if (isErrorWithType(error) && error.body) {
```

Comment:
On second thought, a unit test to cover this change would be helpful.
Could you please add one?